### PR TITLE
fix: Add docs for Smart Contracts 

### PR DESCRIPTION
**Pull request by nakul1010**
- [x] Add docs for smart contracts. 
- [x] Add a py script that formats the smart contract doc according to docusaurus. 
- [x] integrate it with github ci

Steps
- Contratct docs build command `forge doc --out docs/docs/contracts`
- Script to refactor forge docs with docusaurus `python3 docs_reformat_scripts/forge_doc_reformat.py`
- Build docusaurus, command is mentioned in vercel [here](https://vercel.com/interlay/bob/settings)

Note: 
- Temporary was blocked by [foundry issue #6386](https://github.com/foundry-rs/foundry/issues/6386).

### CODE REVIEW

File: docs/docs/contracts/src/src/bridge/BitcoinTx.sol/library.BitcoinTx.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# BitcoinTx
```
Comment (nud3l) [inline docs/docs/contracts/src/src/bridge/BitcoinTx.sol/library.BitcoinTx.md:1]: @nakul1010 are these files supposed to be committed? I'd imagine these to be just build files that are generated on the fly?

Comment (nakul1010) [reply docs/docs/contracts/src/src/bridge/BitcoinTx.sol/library.BitcoinTx.md:1]: yaa it shouldn't be missed it in `gitignore`

**Comment by nakul1010** [comment]
@nud3l & @gregdhill need to push generated `forge docs` can't `gitignore` it otherwise docusaurus won't pick it up. But it will only commit if there are any changes to the smart contracts. 
Ideally need to design and maintain plugin similar to what hardhat has [here](https://github.com/primitivefinance/primitive-dodoc)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -37,3 +38,14 @@ jobs:
         run: |
           forge test -vvv
         id: test
+
+      - name: Run Forge docs
+        run: |
+          forge doc --out docs/docs/contracts
+          python3 docs_reformat_scripts/forge_doc_reformat.py
```
Comment (nud3l) [inline .github/workflows/test.yml:45]: How about putting this into a `/docs/scripts/` instead of creating a top level folder?

**Review by nud3l — APPROVED** [review]

**Pull request merged**